### PR TITLE
CIV-9112 RPA for Mediation in Lip Claimant Response

### PR DESCRIPTION
**Pull request by kenneth-hmcts**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-9112

### Change description ###
Set Up RPA for Lip Claimant Response

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by neeta-hmcts — DISMISSED** [review]

**Review by neeta-hmcts — COMMENTED** [review]
plz verify this flow locally too.. as many changes are merged for task list since this ticket is created,

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts [hunk 1]
```diff
@@ -104,6 +104,11 @@ export function buildWhatToDoNextSection(claim: Claim, claimId: string, lang: st
     tasks.push(freeTelephoneMediationTask);
   }
 
+  if (claim?.claimantResponse?.hasPartPaymentBeenAccepted?.option === YesNo.NO && claim.isFullDefence() && claim.responseStatus === ClaimResponseStatus.RC_PAID_FULL) {
```
Comment (jarekPierchala) [inline src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts:107]: please put that condition to separate func

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistService.ts [hunk 2]
```diff
@@ -8,15 +8,16 @@ import {
...
   buildYourResponseSection,
 } from './claimantResponseTasklistBuilder';
+import {ClaimResponseStatus} from 'models/claimResponseStatus';
 
 export function getClaimantResponseTaskLists (claim: Claim, claimId: string, lng: string) {
   const lang = getLng(lng);
   const taskGroups : TaskList[] = [];
   taskGroups.push(buildHowDefendantRespondSection(claim, claimId, lang));
-  if(claim.isPartialAdmissionNotPaid() || (claim.isFullDefence() && claim.isRejectAllOfClaimDispute()) || claim.isFullAdmission()) {
+  if(claim.isPartialAdmissionNotPaid() || (claim.isFullDefence() && (claim.isRejectAllOfClaimDispute() || claim.responseStatus === ClaimResponseStatus.RC_PAID_FULL)) || claim.isFullAdmission()) {
```
Comment (jarekPierchala) [inline src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistService.ts:17]: condition to separate function please

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistService.ts [hunk 3]
```diff
@@ -8,15 +8,16 @@ import {
...
     taskGroups.push(buildWhatToDoNextSection(claim, claimId, lang));
   }
-  if(claim.isPartialAdmissionPaid() || (claim.isFullDefence() && !claim.isRejectAllOfClaimDispute()))
+  if(claim.isPartialAdmissionPaid() || (claim.isFullDefence() && !claim.isRejectAllOfClaimDispute() && claim.responseStatus === ClaimResponseStatus.RC_PAID_LESS))
```
Comment (jarekPierchala) [inline src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistService.ts:20]: same here

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts (hunk 1 above)
Comment (kenneth-hmcts) [reply src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts]: Changes made

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistService.ts (hunk 2 above)
Comment (kenneth-hmcts) [reply src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistService.ts]: Changes made

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistService.ts (hunk 3 above)
Comment (kenneth-hmcts) [reply src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistService.ts]: Changes made

**Review by jarekPierchala — APPROVED** [review]

**Review by ElifDenizEsea — APPROVED** [review]

**Review by kalachandrasekar1 — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Review by jeswanth-hmcts — APPROVED** [review]

**Pull request closed without merging**